lib/librato/metrics/queue.rb

```diff
@@ -118,31 +136,36 @@ def queued
         return {} if @queued.empty?
         globals = {}
         globals[:source] = @source if @source
+        globals[:tags] = @tags if has_tags?
         globals[:measure_time] = @measure_time if @measure_time
+        globals[:time] = @time if @time
         @queued.merge(globals)
       end
 
       # Count of metrics currently queued
       #
       # @return Integer
       def size
-        self.queued.inject(0) { |result, data| result + data.last.size }
+        self.queued.reject { |key| key == :multidimensional }.inject(0) { |result, data| result + data.last.size }
       end
       alias :length :size
 
     private
 
       def check_measure_time(data)
-        if data[:measure_time] < Metrics::MIN_MEASURE_TIME
-          raise InvalidMeasureTime, "Measure time for submitted metric (#{data}) is invalid."
-        end
+        invalid_time =
+          data[:measure_time] && data[:measure_time] < Metrics::MIN_MEASURE_TIME ||
+            data[:time] && data[:time] < Metrics::MIN_MEASURE_TIME
```

Comment:
treating `measure_time` and `time` internally as the same will remove all the complexity here